src/test/java/com/rakovets/course/java/core/practice/string/StringUtilTest.java

```diff
@@ -0,0 +1,355 @@
+    void findCountEnglishLettersTest(String text, int expected) {
+        StringUtil stringUtil = new StringUtil();
+        assertEquals(expected, stringUtil.findCountEnglishLetters(text));
+    }
+
+    static Stream<Arguments> provideArgumentsForFindCountPunctuationSymbol() {
+        return Stream.of(
+                Arguments.of("heey, john", 1),
+                Arguments.of(null, -1),
+                Arguments.of(", , , ! . ?", 6),
+                Arguments.of("Usual text", 0),
+                Arguments.of("", 0),
+                Arguments.of("How are you?", 1)
+        );
+    }
+
+    @ParameterizedTest()
+    @MethodSource("provideArgumentsForFindCountPunctuationSymbol")
+    void findCountPunctuationSymbolTest(String text, int expected) {
+        StringUtil stringUtil = new StringUtil();
+        assertEquals(expected, stringUtil.findCountPunctuationSymbol(text));
+    }
+
+    static Stream<Arguments> provideArgumentsForIsPalindrome() {
+        return Stream.of(
+                Arguments.of("Do geese see God?", true),
+                Arguments.of("de,ed", true),
+                Arguments.of("", true),
+                Arguments.of(null, false),
+                Arguments.of("1991", true),
+                Arguments.of("reek", false)
+
+        );
+    }
+
+    @ParameterizedTest()
+    @MethodSource("provideArgumentsForIsPalindrome")
+    void isPalindromeTest(String text, boolean expected) {
+        StringUtil stringUtil = new StringUtil();
+        assertEquals(expected, stringUtil.isPalindrome(text));
+    }
+
+    static Stream<Arguments> provideArgumentsForSplitString() {
+        return Stream.of(
+                Arguments.of("asdpfd", 3, new ArrayList<>(Arrays.asList("asd", "pfd"))),
+                Arguments.of(null, 3, null),
+                Arguments.of("ki ppit", 3, new ArrayList<>(Arrays.asList("ki ", "ppi", "t"))),
+                Arguments.of(" roll ", 3, new ArrayList<>(Arrays.asList(" ro", "ll "))),
+                Arguments.of("jdfs", 0, null),
+                Arguments.of("jdfs", 7, new ArrayList<>(Arrays.asList("jdfs")))
+        );
+    }
+
+    @ParameterizedTest()
+    @MethodSource("provideArgumentsForSplitString")
+    void splitStringTest(String text, int n, ArrayList<String> expected) {
+        StringUtil stringUtil = new StringUtil();
+        assertEquals(expected, stringUtil.splitString(text, n));
+    }
+
+    static Stream<Arguments> provideArgumentsForFindCountSpace() {
+        return Stream.of(
+                Arguments.of("?   123 43", 3),
+                Arguments.of(" ", 0),
+                Arguments.of(null, -1),
+                Arguments.of(" word one", 2)
+        );
+    }
+
+    @ParameterizedTest()
+    @MethodSource("provideArgumentsForFindCountSpace")
+    void findCountSpaceTest(String text, int expected) {
+        StringUtil stringUtil = new StringUtil();
+        assertEquals(expected, stringUtil.findCountSpace(text));
```

Comment:
Add blank line between WHEN  and THEN
```suggestion
        StringUtil stringUtil = new StringUtil();
        
        assertEquals(expected, stringUtil.findCountSpace(text));
```